### ISSUE TITLE
Creating volume Formatting volume: Error creating filesystem on /dev/rbd0 with cmd: \"mkfs.btrfs %\". Error: Command: [/bin/sh -c mkfs.btrfs /dev/rbd0], Exit status 1

### ISSUE THREAD

**Issue by angry-tony** [contiv/volplugin#278]
Hi.. i'm new to volplugin.
when i ran "docker volume create" to use btrfs for docker volume, i got this error.
but ext4 is ok.
i guess btrfs's options need to change.

```
[cluster-admin@node-01 ~]$ sudo docker volume create -d volplugin --name hoho/foo

Error response from daemon: create hoho/foo: VolumeDriver.Create: Could not determine policy configuration Status was not 200: was 500: "Creating volume Formatting volume: Error creating filesystem on /dev/rbd0 with cmd: \"mkfs.btrfs %\". Error: Command: [/bin/sh -c mkfs.btrfs /dev/rbd0], Exit status 1, Runtime 43.78757ms (exit status 1)\n\tgithub.com/contiv/volplugin/volmaster.(*DaemonConfig).handleCreate.func1 [daemon.go 604]\n\tgithub.com/contiv/volplugin/lock.(*Driver).ExecuteWithMultiUseLock [lock.go 105]\n\tgithub.com/contiv/volplugin/volmaster.(*DaemonConfig).handleCreate [daemon.go 612]\n\tgithub.com/contiv/volplugin/volmaster.(*DaemonConfig).(github.com/contiv/volplugin/volmaster.handleCreate)-fm [daemon.go 72]\n\tgithub.com/contiv/volplugin/volmaster.logHandler.func1 [daemon.go 120]\n\tnet/http.HandlerFunc.ServeHTTP [server.go 1618]\n\tgithub.com/contiv/volplugin/vendor/github.com/gorilla/mux.(*Router).ServeHTTP [mux.go 98]\n\tnet/http.serverHandler.ServeHTTP [server.go 2081]\n\tnet/http.(*conn).serve [server.go 1472]\n\truntime.goexit [asm_amd64.s 1998]\n\tgithub.com/contiv/volplugin/storage/backend/ceph.(*Driver).mkfsVolume [internals.go 65]\n\tgithub.com/contiv/volplugin/storage/backend/ceph.(*Driver).Format [ceph.go 124]\n\tgithub.com/contiv/volplugin/volmaster.(*DaemonConfig).formatVolume [volume.go 83]\n\tgithub.com/contiv/volplugin/volmaster.(*DaemonConfig).handleCreate.func1 [daemon.go 600]\n\tgithub.com/contiv/volplugin/lock.(*Driver).ExecuteWithMultiUseLock [lock.go 105]\n\tgithub.com/contiv/volplugin/volmaster.(*DaemonConfig).handleCreate [daemon.go 612]\n\tgithub.com/contiv/volplugin/volmaster.(*DaemonConfig).(github.com/contiv/volplugin/volmaster.handleCreate)-fm [daemon.go 72]\n\tgithub.com/contiv/volplugin/volmaster.logHandler.func1 [daemon.go 120]\n\tnet/http.HandlerFunc.ServeHTTP [server.go 1618]\n\tgithub.com/contiv/volplugin/vendor/github.com/gorilla/mux.(*Router).ServeHTTP [mux.go 98]\n\tnet/http.serverHandler.ServeHTTP [server.go 2081]\n\tnet/http.(*conn).serve [server.go 1472]\n\truntime.goexit [asm_amd64.s 1998]"
	github.com/contiv/volplugin/volplugin.(*DaemonConfig).requestCreate [utils.go 196]
	github.com/contiv/volplugin/volplugin.(*DaemonConfig).create [handlers.go 255]
	github.com/contiv/volplugin/volplugin.(*DaemonConfig).(github.com/contiv/volplugin/volplugin.create)-fm [volplugin.go 143]
	github.com/contiv/volplugin/volplugin.logHandler.func1 [volplugin.go 181]
	net/http.HandlerFunc.ServeHTTP [server.go 1618]
	github.com/contiv/volplugin/vendor/github.com/gorilla/mux.(*Router).ServeHTTP [mux.go 98]
	net/http.serverHandler.ServeHTTP [server.go 2081]
	net/http.(*conn).serve [server.go 1472]
	runtime.goexit [asm_amd64.s 1998]
```

mkfs.btrfs not work.
```
[cluster-admin@node-01 ~]$ /bin/sh -c mkfs.btrfs /dev/rbd0
usage: mkfs.btrfs [options] dev [ dev ... ]
options:
	-A|--alloc-start START  the offset to start the FS
	-b|--byte-count SIZE    total number of bytes in the FS
	-d|--data PROFILE       data profile, raid0, raid1, raid5, raid6, raid10, dup or single
	-f|--force              force overwrite of existing filesystem
	-l|--leafsize SIZE      deprecated, alias for nodesize
	-L|--label LABEL        set a label
	-m|--metadata PROFILE   metadata profile, values like data profile
	-M|--mixed              mix metadata and data together
	-n|--nodesize SIZE      size of btree nodes
	-s|--sectorsize SIZE    min block allocation (may not mountable by current kernel)
	-r|--rootdir DIR        the source directory
	-K|--nodiscard          do not perform whole device TRIM
	-O|--features LIST      comma separated list of filesystem features
	-U|--uuid UUID          specify the filesystem UUID
	-V|--version            print the mkfs.btrfs version and exit
btrfs-progs v3.19.1
```

**Comment by erikh** [comment]
Hmm. I'll look into this one. We have a test using btrfs too. Sorry to cause you so much trouble!

Here's the policy we use in our tests: https://github.com/contiv/volplugin/blob/master/systemtests/testdata/ceph/fs.json

Let me know if you're doing something different, this is definitely a bug!

**Comment by erikh** [comment]
WOW. Nice catch, this means our test is broken too.

**Comment by erikh** [comment]
Ok, I see what's going on here now. I'm going to keep this open until the error message is clarified.

@angry-tony what you need to do is increase the size of your volume if you're using the fs.json 10MB default. This is too small for btrfs. Works just fine with 100MB volumes.

**Comment by erikh** [comment]
Closing this as the error message was corrected in #290
